packages/verification-client/src/Client.js

```diff
@@ -96,6 +120,201 @@ export default class Client {
+          console.log("Recovered Signer address:", issuerAddr)
+
+          if (issuerAddr == claim.proof.creator){
+            console.log("Signer Address matches Claim Creator attribute")
+          }else{
+            console.error("Signer Address does NOT match Claim Creator attribute, discarding.")
+            issuerAddr=null
+            continue
+          }
+
+          issuerName=await this.resolveAndVerifyIssuerIdentitiy(issuerAddr)
+          if (issuerName != undefined){
+            issuerVerified=true
+          }
+
+          if (claim.scope=="register"){
+            console.log("Is a registration claim!")
+            registered=true
+            revoked=false
+          }else if (claim.scope=="revoke"){
+            console.log("Is revocation claim")
+            revoked=true
+          }else{
+            console.error("Is an unkown claim type, discarding.")
+            continue
+          }
+        }else{
+          console.error("Hashes NOT matching, Claim is NOT for this File, discarding.")
+          continue
+        }
+      }else{
+        console.error("Hashes NOT matching, Event is not for this File, discarding.")
+        continue
+      }
+
+    }
+    console.log("Consolidated Verification Result for File "+hash+":")
+    console.log("IssuerAddr: "+issuerAddr+
+        "\nIssuerName: "+issuerName+
+        "\nIssuerVerified: "+issuerVerified+
+        "\nRegistered: "+registered+
+        "\nExpiry: "+expiry+
+        "\nRevoked: "+revoked)
+
+    let issuer=issuerAddr
+
+    return {expiry, issuer, issuerImg, issuerName, issuerVerified, revoked}
+  }
+
+  async resolveAndVerifyIssuerIdentitiy (hash) {
+    // Get Events for Filehash
+    const identitiyEvents = await this.claimContract.getPastEvents(
+        'Claim', {
+          filter: {file: hash},
+          fromBlock: 0,
+        })
+
+    var aLen = identitiyEvents.length;
+
+    // For Each Event
+    for (var i = 0; i < aLen; i++) {
+
+      //Get Claim Hash from Event
+      let identityHash=identitiyEvents[i].returnValues.file
+      let claimHash=identitiyEvents[i].returnValues.hash
+      console.log(identityHash+" >> "+claimHash)
+
+
+      //Get Claim from endpoint for Claimhash
+      var claim = this.getRawClaim(claimHash)
+
+      //Verify Claim (hashes to claimhash, belongs to file, Has right content/type)
+      if (claim.id === "cert:addr:0x"+identityHash) {
+
+        //Get Issuer Hash from Address from Signature
+        let issuerHash=this.verifySignatureAndGetPubkey(claim, claim.signature)
+
+        //Only Certifaction is allowed to issue ID claims
+        if (issuerHash==this.acceptedIssuerKey){
+          return claim.name
+        }
+      }
+    }
+  }
+
+  async getRawClaim(claimhash){
+    try {
+      var claim
+      console.log("Retrieving Claim from https://api.dev.testnet.certifaction.io/claim/"+claimhash)
+      const res =  await axios.get(`https://api.dev.testnet.certifaction.io/claim/${claimhash}`)
```

Comment:
This base url should also be passed like the providerUrl

Reply:
Done in f72b10b7e777f35c4fee40319d92682a0d190b0b